src/Microsoft.DotNet.MSBuildSdkResolver/FXVersion.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.MSBuildSdkResolver
+{
+    // Note: This is not SemVer (esp., in comparing pre-release part, fx_ver_t does not
+    // compare multiple dot separated identifiers individually.) ex: 1.0.0-beta.2 vs. 1.0.0-beta.11
```

Comment:
Unclear what fx_ver_t is in this context, we should point to it in core-setup via link and make note that the behavior of this class must match fx_ver_t.